### PR TITLE
fix: Update disruptions that aren't ready

### PR DESCRIPTION
**Pull request by losvedir**
#### Summary of changes
**Asana Ticket:** [🐛 Unable to edit disruption in Arrow](https://app.asana.com/0/281253957434160/1198920748926221)

The fix is pretty small. Basically, the `update` controller action used `DisruptionRevision.only_ready()` to select the latest "ready" revision to clone when updating. However, I think actually we should be cloning the latest revision. The frontend flow lets you edit only based off the latest draft. And the current flow causes a crash if a given disruption doesn't have a ready revision yet, but is updated (i.e.: two drafts in a row before any approvals).

#### Reviewer Checklist
- [ ] Meets ticket's acceptance criteria
- [ ] Any new or changed functions have typespecs
- [ ] Tests were added for any new functionality (don't just rely on Codecov)
- [ ] This branch was deployed to the staging environment and is currently running with no unexpected increase in warnings, and no errors or crashes.

### CODE REVIEW

File: test/arrow_web/controllers/api/disruption_controller_test.exs [hunk 1]
```diff
@@ -532,6 +532,104 @@ defmodule ArrowWeb.API.DisruptionControllerTest do
...
+      disruption_revision =
+        insert(:disruption_revision,
+          disruption: disruption,
+          start_date: ~D[2019-10-10],
+          end_date: ~D[2019-11-01],
+          exceptions: [build(:exception, excluded_date: ~D[2019-10-20])],
+          days_of_week: [
+            build(:day_of_week,
+              day_name: DayOfWeek.date_to_day_name(~D[2019-10-10]),
+              start_time: ~T[20:30:00]
```
Comment (digitalcora) [inline test/arrow_web/controllers/api/disruption_controller_test.exs:555]: This test is a bit hard to follow (100 lines!) — I think partly because it includes a ton of details that ultimately aren't relevant to the behavior being tested. Maybe we can just pick one attribute to update for this test, like the end date, and strip out everything else that isn't strictly required? Both from the factory calls, and the post data.

Comment (losvedir) [reply test/arrow_web/controllers/api/disruption_controller_test.exs]: Good idea. I stripped out some of the unnecessary data (and, added a bit more testing, so the actual lines of code is the same or longer, but more relevant).

File: test/arrow_web/controllers/api/disruption_controller_test.exs [hunk 2]
```diff
@@ -532,6 +532,80 @@ defmodule ArrowWeb.API.DisruptionControllerTest do
...
+        "data" => %{
+          "type" => "disruption",
+          "id" => disruption.id,
+          "attributes" => %{
+            "start_date" => Date.to_iso8601(disruption_revision.start_date),
+            "end_date" => Date.to_iso8601(~D[2019-12-01])
+          },
+          "relationships" => %{
+            "days_of_week" => %{
+              "data" => [%{"type" => "day_of_week", "attributes" => %{"day_name" => "saturday"}}]
```
Comment (digitalcora) [inline test/arrow_web/controllers/api/disruption_controller_test.exs:567]: Do we need to include all the existing (unchanged) attributes in the update? I don't think we _should_ need to, based on how Ecto updates typically work, and that could improve the readability a bit.

Comment (losvedir) [reply test/arrow_web/controllers/api/disruption_controller_test.exs]: Huh, I had thought if we didn't include them that would tell it to *delete* the associations, but from some playing around it looks like that's only if you include the relationship but set `"data" => []` which makes sense. I removed the extraneous attributes.

**Review by digitalcora — APPROVED** [review]
⏱️✨
